### ISSUE TITLE
Quantille Loss

### ISSUE THREAD

**Issue by KonradRtz** [Time-MoE/Time-MoE#64]
Dear Authors,
Thank you for your brilliant work, could you maybe add the code infrastructure to support fine-tuning for quantille regression with the pin-ball loss-function, additionally I would also love to see the Fissler-Ziegel loss function. However quantille regression would be especially interesting, since for example there has been experiments about quantille regression for TimesFM.

Thank you for your work.

Kind regards

**Comment by Maple728** [comment]
Thank you for your interest in our work!

Replacing the Huber loss with your desired loss function is straightforward. In the [model file](https://github.com/Time-MoE/Time-MoE/blob/main/time_moe/models/modeling_time_moe.py#L963), modify the following line:

```python
self.loss_function = torch.nn.HuberLoss(reduction='none', delta=2.0)
```

Change it to your preferred loss function, like this:

```python
self.loss_function = quantile_loss_function
```

Please feel free to contact us, if you have any questions.
